lib/AL_USDMaya/AL/usdmaya/Global.cpp

```diff
@@ -45,12 +58,165 @@ namespace {
   // once per top-level file read operation (ie, once per open, or once
   // per import, or once per reference).
   std::atomic<size_t> readDepth;
+
+#if defined(WANT_UFE_BUILD)
+  // The proxy shape has an internal cache which needs to update when any of
+  // its UFE scene items are selected and transformed.
+  class UfeTransformObserver : public Ufe::Observer
+  {
+  public:
+    UfeTransformObserver() : Ufe::Observer()
+    {
+    }
+
+    ~UfeTransformObserver() override
+    {
+    }
+
+    void operator()(const Ufe::Notification& notification) override
+    {
+      auto xformChanged = dynamic_cast<const Ufe::Transform3dChanged*>(&notification);
+      if (xformChanged == nullptr) return;
+
+      // Action for USD scene items only.
+      Ufe::SceneItem::Ptr sceneItem = xformChanged->item();
+      if (!sceneItem || (sceneItem->runTimeId() != AL::usdmaya::USD_UFE_RUNTIME_ID)) return;
+
+      std::string mayaPath = sceneItem->path().popSegment().popHead().string();
+
+      MSelectionList sl;
+      sl.add(MString(mayaPath.c_str()));
+
+      MObject object;
+      MStatus status = sl.getDependNode(0, object);
+      if (!status) return;
+
+      MFnDependencyNode dependNode(object, &status);
+      if (!status) return;
+
+      if (dependNode.typeId() == AL::usdmaya::nodes::ProxyShape::kTypeId)
+      {
+        auto proxyShape =
+          dynamic_cast<AL::usdmaya::nodes::ProxyShape*>(dependNode.userNode());
```

Comment:
This should be a static_cast (since you have already validated via the MTypeID that the node is of the correct type, so the extra overhead of the RTTI lookups of dynamic_cast is not needed at all here - static_cast will be zero cost in this case)

Reply:
Agreed. Change made.